### ISSUE TITLE
Stacks can´t add_candidates

### ISSUE THREAD

**Issue by gustavomodelli** [tidymodels/stacks#48]
## The problem

I'm having trouble with stacks in add candidates. I had the error message:
Erro: Element `attr(data_stack, "train")` needs to inherit from `tbl_df`, but its class is `data.frame`.
Run `rlang::last_error()` to see where the error occurred.

## Reproducible example

```r
library(tidyverse)
library(arsenal)
library(lubridate)
library(tidymodels)
library(vip)
library(stacks)

durval <- read.csv('durval_final.csv', stringsAsFactors = FALSE,
                   sep = ';', dec = ',')

durval_anal <- durval %>% 
  select(idade, sc_receptor, cor, sexo, tipo_dialise, d_base, diabetes,
         PRA_I, re_tx, mismatchs, idade_doador, sexo_doador,
         cor_doador, imc_doador, causa_morte, has_doador, dm_doador,
         Na_doador, cpk_doador, creat_doador_final, DVA, TIF,
         tempo_anastomose, inducao, DGF)


## Split Data -----------------------------------------------------

set.seed(123)
split <- initial_split(durval_anal, prop = 0.75)
durval_train <- training(split)
durval_test <- testing(split)

## -- Analise Recipe ---------------------------------------------

rec_lasso <- recipe(DGF ~ . , data = durval_train) %>% 
  step_rm(sexo, sexo_doador, cor, cor_doador, re_tx, mismatchs, d_base, tipo_dialise, inducao) %>% 
  step_BoxCox(all_numeric()) %>% 
  step_medianimpute(all_numeric()) %>% 
  step_center(all_numeric()) %>% 
  step_scale(all_numeric()) %>% 
  step_corr(all_numeric(), threshold = 0.9) %>% 
  step_ns(TIF, creat_doador_final, deg_free = 4) %>% 
  step_dummy(all_nominal(), -all_outcomes()) %>% 
  step_nzv(all_predictors())

rec_tree <- recipe(DGF ~ . , data = durval_train) %>% 
  step_rm(sexo, sexo_doador, cor, cor_doador, re_tx, mismatchs, d_base, tipo_dialise, inducao) %>% 
  step_medianimpute(all_numeric()) %>% 
  step_dummy(all_nominal(), -all_outcomes(), one_hot = TRUE) %>% 
  step_nzv(all_predictors())


## --- Model -----------------------------------------------------

model_lasso <- logistic_reg(penalty = tune(), mixture = tune()) %>% 
  set_engine('glmnet')

model_xgb <- boost_tree(mode = 'classification', mtry = tune(), min_n = tune(), tree_depth = tune(), loss_reduction = tune()) %>% 
  set_engine('xgboost')


## --- workflow ---------------------------------------------------

work_lasso <- workflow() %>% 
  add_recipe(rec_lasso) %>% 
  add_model(model_lasso)

work_xgb <- workflow() %>% 
  add_recipe(rec_tree) %>% 
  add_model(model_xgb)

## --- folds -----------------------------------------------------

set.seed(123)
folds <- bootstraps(durval_train, times = 10)

## --- tunes ---------------------------------------------------

doParallel::registerDoParallel()

set.seed(123)
tune_lasso <- tune_grid(
  work_lasso,
  resamples = folds,
  grid = 15,
  metrics = metric_set(roc_auc, accuracy),
  control = control_stack_grid()
)

set.seed(123)
tune_xgb <- tune_grid(
  work_xgb,
  resamples = folds,
  grid = 10,
  metrics = metric_set(roc_auc, accuracy),
  control = control_stack_grid()
)


stack01 <- stacks() %>% 
  add_candidates(tune_lasso) %>% 
  add_candidates(tune_xgb)
```

**Comment by simonpcouch** [comment]
Hi there! I believe this was fixed in #46. Are you still seeing this error with the most recent dev version?

**Comment by gustavomodelli** [comment]
Thank you for your quick response, Simon.
I have downloaded the latest version of stacks and fix the problem.
